### PR TITLE
Selinux enforcing support for RHEL/Centos

### PR DESCRIPTION
**Pull request by AnMoeller**
lets have an attribute for enforcing or permissiving SELinuxon RHEL/Centos.
Implementation like suggested in issue: https://github.com/dev-sec/chef-os-hardening/issues/106

### CODE REVIEW

File: recipes/selinux.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# SELinux enforcing support
+
+case node['platform_family']
+when 'rhel', 'fedora'
+  unless node['os-hardening']['security']['selinux_mode'] == 'unmanaged'
+    if node['os-hardening']['security']['selinux_mode'] == 'enforcing'
```
Comment (artem-sidorenko) [inline recipes/selinux.rb]: Suggestion:
```ruby
unless ...
  semode = case node['os-hardening']['security']['selinux_mode']
                    when 'enforcing'
                      'Enforcing'
                    when 'permissive'
                      'Permissive'
                     else
                       raise "Unsupported selinux mode #{node['os-hardening']['security']['selinux_mode']}"
                    end

  execute "Set selinux mode to #{semode}" do
    command "setenforce #{semod}"
    not_if "getenforce | grep -F '#{semode}'"
  end

...
end
```

File: recipes/selinux.rb [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    elsif node['os-hardening']['security']['selinux_mode'] == 'permissive'
+      execute 'setenforce Permissive' do
+        command 'setenforce Permissive'
+        not_if "getenforce | grep -F 'Enforcing'"
+      end
+    end
+
+    template '/etc/selinux/config' do
+      source 'rhel_selinuxconfig.erb'
+      mode 0544
```
Comment (artem-sidorenko) [inline recipes/selinux.rb]: unsure, do we need 0`5`44 here? not 0`6`44 ?

File: README.md [hunk 3]
```diff
@@ -95,6 +95,8 @@ It will not:
   * ypserv ([NSA](http://www.nsa.gov/ia/_files/os/redhat/rhel5-guide-i731.pdf), Chapter 3.2.4)
   * telnet-server ([NSA](http://www.nsa.gov/ia/_files/os/redhat/rhel5-guide-i731.pdf), Chapter 3.2.2)
   * rsh-server ([NSA](http://www.nsa.gov/ia/_files/os/redhat/rhel5-guide-i731.pdf), Chapter 3.2.3)
+* `default['os-hardening']['security']['selinux_mode'] = 'unmanaged'`
```
[the remark this answers is not in the document]
Comment (AnMoeller) [reply README.md]: done

File: recipes/selinux.rb (hunk 1 above)
Comment (AnMoeller) [reply recipes/selinux.rb]: tanks for this suggestion. So it is easier to read.

**Comment by AnMoeller** [comment]
Many thanks for your hints and suggestions. They are implemented.

**Comment by artem-sidorenko** [comment]
Full CI test is available here: https://travis-ci.org/artem-forks/chef-os-hardening/builds/290410563

**Comment by artem-sidorenko** [comment]
@atomic111 there is some issue with DO image name (centos 7), I'll fix it in another PR today. When we probably can rebase this PR

**Comment by artem-sidorenko** [comment]
It looks like the issue with centos DO image was already fixed in master, so I'm merging this.

@AnMoeller many thanks!

**Pull request merged**